widgetssdk/src/main/java/com/glia/widgets/core/configuration/EngagementConfiguration.kt

```diff
@@ -6,121 +6,30 @@ import com.glia.widgets.GliaWidgets
+        _runTimeTheme = intent.getParcelableExtraCompat(GliaWidgets.UI_THEME) ?: Dependencies.sdkConfigurationManager.uiTheme,
+        _screenSharingMode = intent.getSerializableExtraCompat(GliaWidgets.SCREEN_SHARING_MODE)
+            ?: Dependencies.sdkConfigurationManager.screenSharingMode
+    )
 
-internal class EngagementConfiguration private constructor(builder: Builder) {
-    val companyName: String?
-    val queueIds: List<String>?
-    val contextAssetId: String?
-    private val contextUrl: String?
     val runTimeTheme: UiTheme?
-        get() = field ?: Dependencies.sdkConfigurationManager.uiTheme
-    val screenSharingMode: ScreenSharing.Mode?
-        get() = field ?: Dependencies.sdkConfigurationManager.screenSharingMode
-    val chatType: ChatType?
-    private val manualStringOverrideL: String?
-
-    init {
-        companyName = builder.companyName
-        queueIds = builder.queueIds
-        contextAssetId = builder.contextAssetId
-        contextUrl = builder.contextUrl
-        runTimeTheme = builder.runTimeTheme
-        screenSharingMode = builder.screenSharingMode
-        chatType = builder.chatType
-        manualStringOverrideL = builder.manualLocaleOverride
-    }
-
-    @Deprecated("")
-    fun getContextUrl(): String? {
-        logDeprecatedMethodUse(TAG, "getContextUrl()")
-        return contextUrl
-    }
-
-    class Builder {
-        var companyName: String? = null
-        var queueIds: List<String>? = null
-        var contextAssetId: String? = null
-        var contextUrl: String? = null
-        var runTimeTheme: UiTheme? = null
-        var screenSharingMode: ScreenSharing.Mode? = null
-        var chatType: ChatType? = null
-        var manualLocaleOverride: String? = null
-        fun companyName(companyName: String?): Builder {
-            this.companyName = companyName
-            return this
-        }
-
-        @Deprecated("")
-        fun queueId(queueId: String?): Builder {
-            this.queueIds = queueId?.let { listOf(queueId) }
-            return this
-        }
-
-        fun queueIds(queueIds: List<String>?): Builder {
-            this.queueIds = queueIds
-            return this
-        }
-
-        @Deprecated("")
-        fun contextUrl(contextUrl: String?): Builder {
-            logDeprecatedMethodUse(TAG, "contextUrl(String)")
-            this.contextUrl = contextUrl
-            return this
-        }
+        get() = _runTimeTheme ?: Dependencies.sdkConfigurationManager.uiTheme
 
-        fun contextAssetId(contextAssetId: String?): Builder {
-            this.contextAssetId = contextAssetId
-            return this
-        }
-
-        @Deprecated(
-            "While UiTheme can still be used for UI customization, we strongly encourage adopting remote configurations(GliaWidgetsConfig.Builder.setUiJsonRemoteConfig). " +
-                "The remote configurations approach is more versatile and better suited for future development."
-        )
```

Comment:
I think we should keep this deprecation notice, shouldn't we?

Reply:
No, I removed it because this is an internal class. I didn't touch any public deprecated methods.
Since these features will be released in version 3.0, we should recheck and remove any remaining deprecations. However, this should only be done after completing the Navigator approach, as several public interfaces that contain deprecations will also be removed.